src/System.Private.CoreLib/shared/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -690,13 +690,12 @@ public static string GenerateProgIdForType(Type type)
                 throw new ArgumentException(SR.Argument_NeedNonGenericType, nameof(type));
             }
 
-            IList<CustomAttributeData> cas = CustomAttributeData.GetCustomAttributes(type);
-            for (int i = 0; i < cas.Count; i++)
+            foreach (CustomAttributeData cad = type.GetCustomAttributesData())
```

Comment:
FWIW, this still has two interface calls for every element (MoveNext and Current on the enumerator).  If it were instead:
```C#
IList<CustomAttributeData> cas = CustomAttributeData.GetCustomAttributes(type);
int count = cas.Count;
for (int i = 0; i < count; i++)
{
    CustomAttributeData cad = cas[i];
    ...
}
```
it'd drop to one for each element.

Reply:
That's the usual trade-off many C# features forces you to make: Write nice readable code; or write fast code.

Since this method is not perf critical, I would opt for nice readable code.

Reply:
> I would opt for nice readable code.

That's fine.  Though personally I'm not sure:
```C#
foreach (CustomAttributeData cad in CustomAttributeData.GetCustomAttributes(type))
{
    …
}
```
is so much more readable than:
```C#
IList<CustomAttributeData> cas = CustomAttributeData.GetCustomAttributes(type);
int count = cas.Count;
for (int i = 0; i < count; i++)
{
    CustomAttributeData cad = cas[i];
    ...
}
```
especially since the … is so much code.  But, as you say, not perf critical, and plenty of other expenses here that would be more problematic if it were.